homeassistant/components/recorder/migration.py

```diff
@@ -169,6 +170,22 @@ class _ColumnTypesForDialect:
 }
 
 
+def _unindexable_legacy_column(
+    instance: Recorder, base: type[DeclarativeBase], err: Exception
+) -> bool:
+    """Ignore index errors on char(0) columns."""
+    # 1167: The used storage engine can't index column '%s'
+    return bool(
+        base == LegacyBase
+        and isinstance(err, OperationalError)
+        and instance.engine
+        and instance.engine.dialect.name == SupportedDialect.MYSQL
+        and isinstance(err.orig, BaseException)
+        and err.orig.args
+        and err.orig.args[0] == 1167
```

Comment:
https://github.com/PyMySQL/mysqlclient/blob/207ac1a948a3e4ae31ceadb78d08bf83bd2f2520/src/MySQLdb/constants/ER.py#L191 there is a constant for this ... but probably not a good way to get it

Reply:
Right, we'd then have to locally import the constant after checking if the supported dialect is `SupportedDialect.MYSQL`, if it isn't they probably don't have `PyMySQL` installed.

I could add a comment explaining that's why we hard code the number here?

Reply:
> I could add a comment explaining that's why we hard code the number here?

Works for me